### PR TITLE
add: time travel into sandbox

### LINKED ISSUE #1: Display AST at all steps of transformation (time travel)
From sandboxes created by [ianjennings](https://github.com/ianjennings): babel/sandboxes#17  

- [ ] show AST at all steps in the transformation (babel-time-travel)
    - archived repo: https://github.com/babel/babel-time-travel
    - PR to add to current repl https://github.com/babel/website/pull/1736

### PR DESCRIPTION
**Pull request by wlawt**
closes #1 

What is does currently:
- Able to get the history of transformations on the source -> compiled output
- Able to display the history as cards show at the bottom

TODO: 
- [x] Add hover over cards to change output field to the code in the card

image

### CODE REVIEW

**Comment by ianjennings** [comment]
Can you please isolate the time travel component into it's own module? This will help us keep cadence while we merge.

**Comment by wlawt** [comment]
## PR is ready to be reviewed!
**Some things to note:**
- TimeTravel has its own component now in `TimeTravel.js` 
- The logic works as following:
    - From CompiledOutput, we pass data down to the child 
    - Based on mouse events (enter/exit) of the card, we send data back up to the parent
    - We then display the card hovered in Result.js tab 
- By default, onLoad the Result.js tab is the source (input) itself
- By default, once `env` preset is turned on, Result.js will be updated to the compiled and will be the ground truth
    - *If you unhover, it'll return to the compiled code

**Let me know, if the contents inside the card should change, or the format of it!** 

DEMO: https://www.loom.com/share/25170e18041a4d80be14f81c4b56faf4

File: src/components/App.js [hunk 1]
```diff
@@ -20,11 +20,11 @@ window.babel = Babel;
  */
 export function convertToBabelConfig(jsonConfig) {
   let result = { plugins: [], presets: [] };
-  result.plugins = jsonConfig.plugins?.map(plugin => [
+  result.plugins = jsonConfig.plugins ?.map(plugin => [
```
Comment (garrettluu) [inline src/components/App.js:23]: I guess this is a linter thing?

File: src/components/Input.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Grid, Menu, Segment } from "semantic-ui-react";
 import { Code } from "./styles";
 
 export function Input({ size, gzip, source, setSource }) {
+  // console.log(timeTravel)
```
Comment (garrettluu) [inline src/components/Input.js]: Remove before merge

File: src/components/Output.js [hunk 3]
```diff
@@ -11,14 +11,16 @@ export function Output({
...
+      <Fragment>
+        <CompiledOutput
+          source={debouncedSource}
+          customPlugin={enableCustomPlugin ? customPlugin : undefined}
+          config={config}
+          key={index}
+          onConfigChange={config => updateBabelConfig(config, index)}
+          removeConfig={() => removeBabelConfig(index)}
+        />
+      </Fragment>
```
Comment (garrettluu) [inline src/components/Output.js]: Why does this need to be wrapped in a `Fragment`?

**Review by garrettluu — APPROVED** [review]
Good job on the time travel! 😄

File: src/components/App.js (hunk 1 above)
Comment (wlawt) [reply src/components/App.js:23]: Yeah, onSave keeps doing that

File: src/components/Output.js (hunk 3 above)
Comment (wlawt) [reply src/components/Output.js]: Ah thanks for reminding! I had previously moved TimeTravel out to be siblings with CompiledOutput, so Fragment would just couple them without producing a `<div>`-like tag, I'll remove it

File: src/components/Transitions.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        });
+    };
+
+    wrapPluginVisitorMethod = (
+        pluginAlias, // : any
+        visitorType, // : any
+        callback // : any
+    ) => {
+        return (...args) => { // : any
+            // $FlowFixMe
```
Comment (barronwei) [inline src/components/Transitions.js]: Is this using Flow?

File: src/standalone.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 // take from @babel/standalone
-import {availablePlugins, availablePresets} from '@babel/standalone';
+import { availablePlugins, availablePresets } from '@babel/standalone';
 
 export function transpilePlugin(pluginString, presets) {
   return Babel.transform(pluginString, {
     babelrc: false,
     configFile: false,
-    ast: false,
+    ast: true,
```
Comment (barronwei) [inline src/standalone.js]: What does this option do?

File: src/components/Transitions.js (hunk 4 above)
Comment (wlawt) [reply src/components/Transitions.js]: Nope, removed all the flow, left the comments just for reference, but will remove

File: src/standalone.js (hunk 5 above)
Comment (wlawt) [reply src/standalone.js]: From testing, it doesn't seem to do anything different in terms of this PR's feature

**Review by barronwei — APPROVED** [review]
Looks good 💯

**Pull request merged**